srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+.. _section-custom-types:
+
+============
+Custom Types
+============
+
+*RTI Perftest* allows the user to add their own `idl` types, instead of using
+the one provided by default. The purpose of this feature is to get a closer
+approximation to the performance behavior of the user's final system. The main
+reason why using one type or another will result in different outputs for latency
+and throughput is the serialization and deserialization times.
+
+This feature is designed to work for regular data, *DynamicData*
+and *FlatData* types.
+
+**Note 1:** For *DynamicData*, the way this is implemented is by representing the data
+as a regular *idl* generated type, and then, internally use the *DynamicData*
+type support representation equivalent for that type as well as build the samples
+with the *DynamicData* API. Hence, providing the *idl* structure is required to the user.
+
+**Note 2:** The use of *Flat Data* requires the generation of a regular type as
+well due to the current implementation. See especific notes in a dedicated section
+below.
```

Comment:
**Note 2:** To test *Flat Data* serialization/deserialization of user data types requires the user to define both a regular datatype and a *FlatData* variant.  See specific notes in the section **Custom Types + `FlatData`** below.